Adding non mandatory fields in ProtectedItem and AzureVmWorkloadProte…

SanjayHukumRana opened the pull request:
…ctionPolicy

Adding additional fields for ProtectedItem related to soft delete feature ; description of each field is in the fields.
In AzureVmWorkloadProtectionPolicy adding another field for fixup of policy

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

SanjayHukumRana committed:
Adding non mandatory fields in ProtectedItem and AzureVmWorkloadProtectionPolicy

Adding additional fields for ProtectedItem related to soft delete feature ; description of each field is in the fields.
In AzureVmWorkloadProtectionPolicy adding another field for fixup of policy

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/Azure/azure-rest-api-specs?pullRequest=6818) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/Azure/azure-rest-api-specs?pullRequest=6818) before we can accept your contribution.<br/><br/>:x: SanjayHukumRana [sign now](https://cla.opensource.microsoft.com/Azure/azure-rest-api-specs?pullRequest=6818)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/Azure/azure-rest-api-specs?pullRequest=6818) it.</sub>

azuresdkci commented:
Can one of the admins verify this patch?

AutorestCI commented:
# Automation for azure-sdk-for-python

AutorestCI commented:
# Automation for azure-sdk-for-go

SanjayHukumRana committed:
Removing redundant fields from child class MabFileFolderProtectedItem

Since these fields are added in base class ProtectedItem , removing them from MabFileFolderProtectedItem

SanjayHukumRana committed:
Rehydrate removal from description

Rehydrated is not in dictionary of swagger valiation  and thus removed from description

ravbhatnagar commented on specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2016-12-01/bms.json:
```diff
@@ -5417,16 +5421,7 @@
           "description": "Protected, ProtectionStopped, IRPending or ProtectionError",
           "type": "string"
-        },
-        "isScheduledForDeferredDelete": {
```
this is a breaking change - removing existing properties.

SanjayHukumRana replied:
The field is moved to base class ProtectedItem and thus not a breaking change

ravbhatnagar commented on specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2017-07-01/bms.json:
```diff
@@ -1745,6 +1745,10 @@
           "items": {
             "$ref": "#/definitions/SubProtectionPolicy"
           }
+        },
```
adding new writable properties requires a new api-version.

TAdityaAnirudh replied:
Hi. Can you please elaborate on why it needs new api-version? This doesn't break existing clients because this is an optional parameters and null is the default value.

ravbhatnagar replied:
https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#123-definition-of-a-breaking-change
paragraph 2 sentence 2. We are just trying to enforce what the architects have laid out as definition of a breaking change for Azure.

SanjayHukumRana replied:
This change is removed in this PR and will be added with new api version in later PR.

ravbhatnagar reviewed (commented):
@SanjayHukumRana - few comments.

SanjayHukumRana committed:
Updating the Field for deferred deleted Time

Updating the new field for deleted time in ProtectedItem while keeping the syncTime for MABfileFolderProtectedItem

SanjayHukumRana committed:
Undo change to AzureVmWorkloadProtectionPolicy

Removed the change to AzureVmWorkloadProtectionPolicy as this requires new api version

ravbhatnagar commented:
Signing off from ARM side.

SanjayHukumRana committed:
Fixing camelcasing

Camelcasing of  isDeferredDeleteScheduleUpcoming field
